File: app/src/main/java/DBHandler.java
import android.content.ContentValues;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class DBHandler extends SQLiteOpenHelper {

    private static final String DB_NAME = "passwordDB";
    private static final int DB_VERSION = 1;
    private static final String TABLE_NAME = "passwords";
    private static final String TITLE_COL = "pwTitle";
    private static final String ID_COL = "pwID";
    private static final String DESC_COL = "pwDesc";
    private static final String PW_COL = "password";



    // constructor for database handler
    public DBHandler(Context context) {
        super(context, DB_NAME, null, DB_VERSION);
    }

    // method for creating a database by running a sqlite query
    @Override
    public void onCreate(SQLiteDatabase db) {
        // create sqlite query, set col names and type
        String query = "CREATE TABLE " + TABLE_NAME + " ("
                + ID_COL + " INTEGER PRIMARY KEY AUTOINCREMENT, "
                + TITLE_COL + " TEXT,"
                + DESC_COL + " TEXT,"
                + PW_COL + " TEXT)";
        // method to execute above sql query
        db.execSQL(query);
    }

    // method to add new pw to sqlite database
    public void addNewCourse(String pwTitle, String pwID, String pwDesc, String password) {

        // create variable for db and call writable to write
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues values = new ContentValues();

        // pass all values with kv pair
        values.put(TITLE_COL, pwTitle);
        values.put(ID_COL, pwID);
        values.put(DESC_COL, pwDesc);
        values.put(PW_COL, password);

        // pass values to table
        db.insert(TABLE_NAME, null, values);
        db.close();
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        // check if the table exists already.
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME);
        onCreate(db);
    }
}

File: app/src/main/java/DBHandler.kt
import android.annotation.SuppressLint
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteException
import android.database.sqlite.SQLiteOpenHelper

class DBHandler(context: Context) : SQLiteOpenHelper(context, DB_NAME, null, DB_VERSION) {

    companion object {
        private const val DB_NAME = "passwordDB"
        private const val DB_VERSION = 1
        private const val TABLE_PASSWORDS = "passwords"
        private const val TITLE_COL = "pwTitle"
        private const val ID_COL = "pwID"
        private const val DESC_COL = "pwDesc"
        private const val PW_COL = "password"
    }

    // method for creating a database by running a sqlite query
    override fun onCreate(db: SQLiteDatabase) {
        // create sqlite query, set col names and type
        val query = "CREATE TABLE $TABLE_PASSWORDS (" +
                "$ID_COL INTEGER PRIMARY KEY AUTOINCREMENT, " +
                "$TITLE_COL TEXT," +
                "$DESC_COL TEXT," +
                "$PW_COL TEXT)"
        // method to execute query
        db.execSQL(query)
    }

    fun addPassword(pw: Password): Long {
        // create variable for db and call writable to write
        val db = this.writableDatabase
        val values = ContentValues()

        // pass all values with kv pair
        values.put(ID_COL, pw.pwID)
        values.put(TITLE_COL, pw.pwTitle)
        values.put(DESC_COL, pw.pwDesc)
        values.put(PW_COL, pw.password)

        // pass values to table
        val success = db.insert(TABLE_PASSWORDS, null, values)
        db.close()
        return success
    }
    @SuppressLint("Range")
    fun getPassword(): List<Password> {
        val pwList: ArrayList<Password> = ArrayList<Password>()
        val selectQuery = "SELECT * FROM $TABLE_PASSWORDS"
        val db = this.readableDatabase
        var cursor: Cursor? = null

        try {
            cursor = db.rawQuery(selectQuery, null)
        } catch (e: SQLiteException) {
            db.execSQL(selectQuery)
            return ArrayList()
        }

        var pwID: Int
        var pwTitle: String
        var pwDesc: String
        var password: String

        if (cursor.moveToFirst()) {
            do {
                pwID = cursor.getInt(cursor.getColumnIndex("id"))
                pwTitle = cursor.getString(cursor.getColumnIndex("name"))
                pwDesc = cursor.getString(cursor.getColumnIndex("email"))
                password = cursor.getString(cursor.getColumnIndex("email"))
                val pw = Password(pwID = pwID, pwTitle = pwTitle, pwDesc = pwDesc, password = password)
                pwList.add(pw)
            } while (cursor.moveToNext())
        }
        return pwList
    }

    fun updatePassword(pw: Password): Int {
        val db = this.writableDatabase
        val contentValues = ContentValues()
        contentValues.put(ID_COL, pw.pwID)
        contentValues.put(TITLE_COL, pw.pwTitle)
        contentValues.put(DESC_COL, pw.pwDesc)
        contentValues.put(PW_COL, pw.password)

        val success = db.update(TABLE_PASSWORDS, contentValues, "id=" + pw.pwID, null)
        db.close()
        return success
    }

    fun deletePassword(pw: Password): Int {
        val db = this.writableDatabase
        val contentValues = ContentValues()
        contentValues.put(ID_COL, pw.pwID)
        val success = db.delete(TABLE_PASSWORDS, "id=" + pw.pwID, null)
        db.close()
        return success
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        // check if the table exists already.
        db.execSQL("DROP TABLE IF EXISTS $TABLE_PASSWORDS")
        onCreate(db)
    }
}
